### PR TITLE
React icons rtl support

### PR DESCRIPTION
**Pull request by tomi-msft**
This PR is the first step in adding auto-mirroring to react-icons. It use `griffel/react` `useTextDirection` hook to detect `rtl` and applies styling based on that.

Part of #553

### CODE REVIEW

File: packages/react-icons/src/utils/useIconState.tsx [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import { FluentIconsProps } from "./FluentIconsProps.types";
 import { makeStyles, mergeClasses } from "@griffel/react";
+import { useTextDirection } from '@griffel/react/TextDirectionContext'
```
Comment (behowell) [inline packages/react-icons/src/utils/useIconState.tsx:3]: We'll need to have this function exported from griffel before we can actually use it.

File: packages/react-icons/src/utils/useIconState.tsx [hunk 2]
```diff
@@ -22,7 +27,7 @@ export const useIconState = <TBaseAttributes extends (React.SVGAttributes<SVGEle
   
     const styles = useRootStyles();
   
-    state.className = mergeClasses(styles.root, state.className);
+    state.className = mergeClasses(styles.root, dir === 'rtl' && styles.rtl, state.className);
```
Comment (behowell) [inline packages/react-icons/src/utils/useIconState.tsx:30]: We need a way to only apply this mirroring behavior to icons that need it. One option is a prop like `autoMirror` that defaults to false. We could then change the default to true for specific icons from some list (like rtl.txt) when generating the React components for the icons.

**Review by behowell — CHANGES_REQUESTED** [review]
Need to resolve the griffel deep import, and add a way to only apply this to certain icons, before merging. Thanks!

**Comment by tomi-msft** [comment]
Closing in favor of a cleaner commit history branch

**Pull request closed without merging**